inc/saiport.h

```diff
@@ -1698,6 +1765,95 @@ typedef enum _sai_port_attr_t
      */
     SAI_PORT_ATTR_SYSTEM_PORT,
 
+    /**
+     * @brief MDIX status to the user
+     *
+     * @type sai_port_mdix_mode_status_t
+     * @flags READ_ONLY
+     */
+    SAI_PORT_ATTR_MDIX_MODE_STATUS,
+
+    /**
+     * @brief Configure MDIX mode
+     *
+     * @type sai_port_mdix_mode_t
+     * @flags CREATE_AND_SET
+     * @default SAI_PORT_MDIX_MODE_AUTO
+     */
+    SAI_PORT_ATTR_MDIX_MODE,
+
+    /**
+     * @brief Configure Auto Neg mode
+     *
+     * @type sai_port_auto_neg_config_mode_t
+     * @flags CREATE_AND_SET
+     * @default SAI_PORT_AUTO_NEG_CONFIG_MODE_DISABLED
+     */
+    SAI_PORT_ATTR_AUTO_NEG_CONFIG_MODE,
```

Comment:
Is this applicable only for the 1000BASE-T ports? Can you please call out what port speeds this attribute is actually relevant?

Reply:
@SathishKumarTR-Dell , thanks, please help to review/sign-off this PR?

Reply:
SAI_PORT_ATTR_AUTO_NEG_MODE